test/object_store/conftest.py

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/python3
+
+import os
+import sys
+import logging
+import pytest
+import shutil
+import tempfile
+from dataclasses import dataclass
+
+# use minio_server
+sys.path.insert(1, sys.path[0] + '/../..')
+from test.pylib.minio_server import MinioServer
+from test.pylib.host_registry import HostRegistry
+from test.pylib.cql_repl import conftest
+
+hosts = HostRegistry()
+
+
+def pytest_addoption(parser):
+    conftest.pytest_addoption(parser)
+    # reserved for tests with real S3
+    s3_options = parser.getgroup("s3-server", description="S3 Server settings")
+    s3_options.addoption('--s3-server-address')
+    s3_options.addoption('--s3-server-port', type=int)
+    s3_options.addoption('--s3-server-bucket')
+
+
+@dataclass
+class S3_Server:
+    address: str
+    port: int
+    bucket_name: str
+
+    async def start(self):
+        pass
+
+    async def stop(self):
+        pass
+
+
+@pytest.fixture(scope="function")
+def ssl(request):
+    yield request.config.getoption('--ssl')
+
+
+@pytest.fixture(scope="function")
+def test_tempdir():
+    # with tempfile.TemporaryDirectory(prefix='scylla-test-') as tempdir:
+    tempdir = tempfile.mkdtemp(prefix='scylla-test-')
+    yield tempdir
+    with open(os.path.join(tempdir, 'log'), 'rb') as log:
+        shutil.copyfileobj(log, sys.stdout.buffer)
+
+@pytest.fixture(scope="function")
+async def tempdir_and_s3_server(pytestconfig, test_tempdir: str):
```

Comment:
I'm not even close an expert in these "fixtures" thing, but I thing that it's not good mixing two -- s3_server and tempdir -- in one

Reply:
the problem is that s3_server also consumes the tempdir. i understand this is not pretty. but i want to keep the existing behavior as much as i can at least in this change. if we can put the minio server's logging message in an other run_dir, we would be able to put them into different fixtures.

Reply:
or, we could use a global variable for sharing the run_dir across the different consumers, namely, minio_server and the test itself, what do you think? IMHO, it trades a bad practice for another. but still..

Reply:
@xemul i think understand what you are after. just not sure if you can accept the proposed solution.

Reply:
I'm not a connoisseur of fixtures in the first place, so the only thing that I dislike here is mixing two unrelated concepts in one method :)